Strict DiscoveryService

rullzer opened the pull request:
* Made strict
* Type hints
* Return types

Signed-off-by: Roeland Jago Douma <roeland@famdouma.nl>

rullzer committed:
Strict DiscoveryService

* Made strict
* Type hints
* Return types

Signed-off-by: Roeland Jago Douma <roeland@famdouma.nl>

juliushaertl commented on lib/private/OCS/DiscoveryService.php:
```diff
@@ -60,7 +61,7 @@ public function __construct(ICacheFactory $cacheFactory,
+	public function discover(string $remote, string $service): array {
 		// Check the cache first
 		$cacheData = $this->cache->get($remote . '#' . $service);
 		if($cacheData) {
```
Regarding the next line, maybe we should add a check here since json_decode might be null which will then cause a type error.

rullzer replied:
Sounds sane I'll add it

rullzer replied:
Done

MorrisJobke commented:
ref (#7392)

rullzer committed:
Add extra check so we don't error out on type

json_decode can return false if we have invalid data.
In that case just assume there was nothing cached

Signed-off-by: Roeland Jago Douma <roeland@famdouma.nl>

MorrisJobke commented:
Waiting for https://drone.nextcloud.com/nextcloud/server/4131
